src/main/java/net/fabricmc/loader/FabricLoader.java

```diff
@@ -127,23 +129,44 @@ public EnvType getEnvironmentType() {
 	 * @return The game instance's root directory.
 	 */
 	@Override
-	public File getGameDirectory() {
+	public Path getGameDir() {
 		return gameDir;
 	}
 
+	@Override
+	@Deprecated
+	public File getGameDirectory() {
+		return getGameDir().toFile();
+	}
+
 	/**
 	 * @return The game instance's configuration directory.
 	 */
 	@Override
-	public File getConfigDirectory() {
-		if (!configDir.exists()) {
-			configDir.mkdirs();
+	public Path getConfigDir() {
+		if (!Files.exists(configDir)) {
+			try {
+				Files.createDirectories(configDir);
```

Comment:
Should this just always try to create directories to avoid race conditions?

Reply:
If old IO creates the dirs, then nio should probably replicate that.